### PR TITLE
qclient.artifact_and_preparation_files

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

File: qp_woltka/tests/test_woltka.py [hunk 1]
```diff
@@ -138,6 +133,7 @@ def test_woltka_to_array_rep82(self):
         with open(merge_fp) as f:
             merge = f.readlines()
 
+        prep_file = f'{out_dir}/prep_info.tsv'
```
Comment (charles-cowart) [inline qp_woltka/tests/test_woltka.py]: I'm not a stickler for using os.path.join() over using '/' but I believe most or all of the qiita code uses the former.

File: qp_woltka/tests/test_woltka.py [hunk 2]
```diff
@@ -237,25 +233,20 @@ def test_woltka_to_array_rep82(self):
     def test_woltka_to_array_wol(self):
         # inserting new prep template
         prep_info_dict = {
-            'SKB8.640193': {'run_prefix': 'S22205_S104_L001_R1'},
-            'SKD8.640184': {'run_prefix': 'S22282_S102_L001_R1'}}
+            'SKB8.640193': {'run_prefix': 'S22205_S104_L001_R'},
```
Comment (charles-cowart) [inline qp_woltka/tests/test_woltka.py:236]: 'R' vs 'R1' is an intentional change, correct?

File: qp_woltka/tests/test_woltka.py [hunk 3]
```diff
@@ -265,6 +256,7 @@ def test_woltka_to_array_wol(self):
         with open(merge_fp) as f:
             merge = f.readlines()
 
+        prep_file = f'{out_dir}/prep_info.tsv'
```
Comment (charles-cowart) [inline qp_woltka/tests/test_woltka.py]: same as previously mentioned with join().

File: qp_woltka/woltka.py [hunk 4]
```diff
@@ -67,29 +63,14 @@ def woltka_to_array(input_files, output, database_bowtie2,
...
-    fwd = sorted(input_files['raw_forward_seqs'])
-    rev = []
-    if 'raw_reverse_seqs' in input_files:
-        rev = sorted(input_files['raw_reverse_seqs'])
     n_files = 1
-    for i, (f, r) in enumerate(zip_longest(fwd, rev)):
+    for i, (k, (f, r)) in enumerate(files.items()):
         if i >= n_files*TASKS_IN_SCRIPT:
             n_files += 1
         with open(f'{output}/sample_details_{n_files}.txt', 'a+') as fh:
```
Comment (charles-cowart) [inline qp_woltka/woltka.py]: join()

File: qp_woltka/woltka.py [hunk 5]
```diff
@@ -171,6 +152,8 @@ def woltka_to_array(input_files, output, database_bowtie2,
 
     # Bowtie2 command structure based on
     # https://github.com/BenLangmead/bowtie2/issues/311
+    preparation_information = f'{output}/prep_info.tsv'
```
Comment (charles-cowart) [inline qp_woltka/woltka.py]: join()

**Review by charles-cowart — CHANGES_REQUESTED** [review]
Everything appears in order except for not using os.path.join().

File: qp_woltka/tests/test_woltka.py (hunk 2 above)
Comment (antgonza) [reply qp_woltka/tests/test_woltka.py:236]: yup

File: qp_woltka/tests/test_woltka.py (hunk 1 above)
Comment (antgonza) [reply qp_woltka/tests/test_woltka.py]: changing all

File: qp_woltka/tests/test_woltka.py (hunk 2 above)
Comment (charles-cowart) [reply qp_woltka/tests/test_woltka.py:236]: ty! will close.

**Review by charles-cowart — APPROVED** [review]
Everything looks good, thanks!

**Pull request merged**